tools/sleigh-lift/src/main.cpp

```diff
@@ -17,35 +17,38 @@ static void PrintUsage(void) {
 
 class InMemoryLoadImage : public LoadImage {
 public:
-  explicit InMemoryLoadImage(unsigned long base_addr)
+  explicit InMemoryLoadImage(uint64_t base_addr)
       : LoadImage("nofile"), base_addr(base_addr) {}
 
-  void SetImageBuffer(std::vector<unsigned char> &&buf) {
+  void SetImageBuffer(std::vector<char> &&buf) {
     assert(image_buffer.empty());
     image_buffer = std::move(buf);
   }
 
   void loadFill(unsigned char *ptr, int size, const Address &addr) override {
     uint8_t start = addr.getOffset();
     for (int i = 0; i < size; ++i) {
-      const int64_t offset = (start + i) - base_addr;
-      ptr[i] = (offset >= 0 && offset < image_buffer.size())
-                   ? image_buffer[offset]
-                   : 0;
+      uint64_t offset = start + i;
+      if (offset >= base_addr) {
+        offset -= base_addr;
+        ptr[i] = offset < image_buffer.size() ? image_buffer[i] : 0;
+      } else {
+        ptr[i] = 0;
+      }
     }
   }
 
   std::string getArchType(void) const override { return "memory"; }
   void adjustVma(long) override {}
 
 private:
-  unsigned long base_addr;
-  std::vector<unsigned char> image_buffer;
+  const uint64_t base_addr;
+  std::vector<char> image_buffer;
 };
 
-static std::vector<unsigned char>
-ParseBytes(const std::string &bytes, unsigned long addr, uint64_t addr_size) {
-  std::vector<unsigned char> buffer;
+static std::vector<char> ParseBytes(const std::string &bytes, uint64_t addr,
```

Comment:
`ParseHexBytes`, make it take a `std::string_view`, and return a `std::string`.

Reply:
d2cb7f0 and 2851125